lms/djangoapps/hera/fragments.py

```diff
@@ -213,41 +239,109 @@ class DashboardPageOutlineFragmentView(CourseOutlineFragmentView):
     View for Dashboard Page
     """
     def render_to_fragment(self, request, course_id=None, page_context=None, **kwargs):
+
+        def get_medal(points):
+            if 90 <= points <= 100:
+                return "platinum"
+            elif 80 <= points <= 89:
+                return "gold"
+            elif 70 <= points <= 79:
+                return "silver"
+            else:
+                return "copper"
+
         course_key = CourseKey.from_string(course_id)
-        course_overview = get_course_overview_with_access(request.user, 'load', course_key, check_if_enrolled=True)
         course = modulestore().get_course(course_key)
 
+        course_grade = CourseGradeFactory().read(request.user, course)
+        courseware_summary = course_grade.chapter_grades.values()
+
         course_block_tree = get_course_outline_block_tree(request, course_id)
         if not course_block_tree:
             return None
 
-        incomplete_subsection = {}
-        popup = False
+        field_data_cache = FieldDataCache.cache_for_descriptor_descendents(
+            course.id, request.user, course, depth=2
+        )
+        course_module = get_module_for_descriptor(
+            request.user, request, course, field_data_cache, course.id, course=course
+        )
+        chapter_module = get_current_child(course_module)
+        section_module = get_current_child(chapter_module)
+        active_block_id = section_module.scope_ids.usage_id.block_id
+        subsection_active = ''
+
+        completed_subsections = {}
+
         for section in course_block_tree.get('children'):
             for subsection in section.get('children', []):
-                if not subsection['complete']:
-                    for unit in subsection.get('children', []):
-                        if unit['complete']:
-                            incomplete_subsection = subsection
-                            popup = True
-                            break
+                if active_block_id == subsection['block_id'] and not subsection_active:
+                    course_block_id = subsection['block_id']
+                    subsection_active = subsection
+                if subsection['complete']:
+                    completed_subsections[subsection.get('block_id')] = {
+                        'display_name': subsection.get('display_name'),
+                        'lms_web_url': subsection.get('lms_web_url', '')
+                    }
 
-        units = incomplete_subsection.get('children', [{}])
-        start_over_url = units[0].get('lms_web_url', '')
+        for chapter in courseware_summary:
+            if not chapter['display_name'] == "hidden":
+                for subsection in chapter['sections']: # the sections in chapter are actually subsection
+                    if course_block_id == subsection.location.block_id:
+                        earned = subsection.all_total.earned
```

Comment:
if we don't get to this condition we will get an error bellow when trying to use this one. please predefine it above